### ISSUE TITLE
Cannot deserialize XMLs persisted using version <1.4.9 which contain references to enum types

### ISSUE THREAD

**Issue by michaelgreenn** [x-stream/xstream#238]
### Related to #150 and [this comment](https://github.com/x-stream/xstream/commit/5526a2bffcd5b2a47a5f00d67fa804a105fffcd4#comments). 

I have some XMLs that were serialized using an older version of XStream. After upgrading to 1.4.9+, they can no longer be deserialized due to an 'Invalid reference' exception being thrown in AbstractReferenceUnmarshaller.java. I understand that this is intentional for immutable types, and that backwards compatibility can be achieved using `xstream.addImmutableType(MyClass.class, true)`.

However, this fix does not seem to work if the XML contains references to **enum types.** 

This is because the isReferenceable method in EnumMapper.java will always return false if the type is an enum:

```
public boolean isReferenceable(final Class<?> type) {
    if (type != null && Enum.class.isAssignableFrom(type)) {
        return false;
    } else {
        return super.isReferenceable(type);
    }
}
```

So adding `xstream.addImmutableType(MyClass.class, true)` would not make any difference.

Is there any way to work around this, or am I missing something?

Thanks!

**Comment by joehni** [comment]
Although I cannot remember that enums were ever non-immutable, the logic to force references at deserialization should work also for enums. I've fixed this for the next 1.4.x release that should happen this month.

In the meanwhile you could register a custom mapper as workaround:

```java
XStream xstream = new XStream() {
    protected MapperWrapper wrapMapper(MapperWrapper next) {
        return new MapperWrapper(next) {
            public boolean isReferenceable(final Class type) {
                if (type == MyCLass.class) {
                    return true;
                } else {
                    return super.isReferenceable(type);
                }
            }
        }
    }
};
```

**Comment by michaelgreenn** [comment]
Great, thank you for the quick response and fix!
